src/client/components/GraphScene.tsx

```diff
@@ -126,21 +131,54 @@ export const dataSelecter = (rawData: any, provider: string): any => {
   }
 }
 
-export default function GraphScene(): JSX.Element {
-  const [data] = useState([])
+export default function GraphScene(props: GraphSceneProps): JSX.Element {
+  const [graphData, setGraphData] = useState([] as any)
+  const location = useLocation<Location>()
+
+  const getData = async (start: string, end: string): Promise<number> => {
+    let data
+    console.time('getData')
+    props.setLoading(true)
+    let response
+    const uri = `${window.location.origin}/v1/getDataAllPartners/?firstDate=${start}&secondDate=${end}`
+    try {
+      response = await fetch(uri)
+      data = await response.json()
+      props.setLoading(false)
+      setGraphData(data)
+    } catch (e) {
+      props.setLoading(false)
```

Comment:
Instead of having `props.setLoading(false)` twice, try adding a `finally` block.